tests/ui/useless_conversion.rs

```diff
@@ -41,12 +41,90 @@ fn test_issue_5833() -> Result<(), ()> {
     Ok(())
 }
 
+fn test_issue_5833_alt1() -> Result<(), ()> {
+    let text = "foo\r\nbar\n\nbaz\n";
+    let mut lines = text.lines();
+    if Some("ok") == lines.into_iter().next() {}
+
+    Ok(())
```

Comment:
Is there a reason, why all of these test functions return a `Result`? It seems like it would be easier and shorter to have no return at all

Reply:
Done. It was iike this so I went on doing it. I fixed all tests concerned by this patch.